src/aggregation/agg_req_with_accessor.rs

```diff
@@ -0,0 +1,126 @@
+//! This will enhance the request tree with access to the fastfield and metadata.
+
+use super::{
+    agg_req::Aggregations, bucket::RangeAggregationReq, Aggregation, BucketAggregationType,
+    MetricAggregation, VecWithNames,
+};
+use crate::{fastfield::DynamicFastFieldReader, SegmentReader, TantivyError};
+use std::collections::HashMap;
+
+//pub type AggregationsWithAccessor = HashMap<String, AggregationWithAccessor>;
+pub type AggregationsWithAccessor = VecWithNames<AggregationWithAccessor>;
+
+/// Aggregation tree with fast field accessors.
+///
+#[derive(Clone)]
+pub enum AggregationWithAccessor {
+    Bucket(BucketAggregationWithAccessor),
+    Metric(MetricAggregationWithAccessor),
+}
+
+impl AggregationWithAccessor {
+    pub fn as_bucket(&self) -> &BucketAggregationWithAccessor {
+        match self {
+            AggregationWithAccessor::Bucket(bucket) => bucket,
+            AggregationWithAccessor::Metric(_) => panic!("wrong aggregation type"),
+        }
+    }
+    pub fn as_metric(&self) -> &MetricAggregationWithAccessor {
+        match self {
+            AggregationWithAccessor::Bucket(_) => panic!("wrong aggregation type"),
+            AggregationWithAccessor::Metric(metric) => metric,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct BucketAggregationWithAccessor {
+    /// In general there can be buckets without fast field access, e.g. buckets that are created
+    /// based on search terms. So eventually this needs to be Option or moved.
```

Comment:
or reorganize the code introducing a BucketAggregationTypeWithAccessor.

Reply:
I don't understand, the struct is already effectively BucketAggregationTypeWithAccessor

Reply:
You missed `Type` in the name. I meant that we could introduce a `BucketAggregationTypeWithAccessor` enum that would contain type specific accessor.